packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -1363,8 +1359,14 @@ export default class InternalModel {
     @private
   */
   updateRecordArrays() {
-    // @ts-ignore: Store is untyped and typescript does not detect instance props set in `init`
-    this.store.recordArrayManager.recordDidChange(this);
+    // TODO: do we want to maintain old API with this flag?
+    if (RECORD_ARRAY_MANAGER_IDENTIFIERS) {
+      // @ts-ignore: Store is untyped and typescript does not detect instance props set in `init`
```

Comment:
This `ts-ignore` is slightly incorrect, it seems we just need to expose recordArrayManager from CoreStore